test/unit/test_inference.py

```diff
@@ -476,6 +476,42 @@ def test_topk_func(batch_size, beam_size, target_vocab_size):
     assert all(mx_values == np_values)
 
 
+@pytest.mark.parametrize("batch_size, beam_size, target_vocab_size, top_n",
+                        [(1, 5, 200, 0),
+                         (5, 5, 200, 0),
+                         (1, 100, 200, 5),
+                         (5, 100, 200, 5)])
+def test_samplek_func(batch_size, beam_size, target_vocab_size, top_n):
+    # arrange scores increasing values from left to right, so the best item is always index 0, next-best 1, and so on
+    scores = mx.nd.array([list(range(1, target_vocab_size + 1)) for x in range(batch_size * beam_size)])
+    # normalize
+    target_dists = mx.nd.broadcast_div(scores, mx.nd.sum(scores, axis=1, keepdims=True))
```

Comment:
[nit] could use the fluent method for sum in this function for brevity: x.sum(axis=1) etc